### DISCUSSION TITLE
How to override tab partial?

### DISCUSSION THREAD

**Discussion by aborruso** [squidfunk/mkdocs-material#6223] [Q&A] [answered]
Hi,
I have created `overrides/partials/toc.html` file and edited the mkdocs settings to set the override folder.

If I write `lorem ipsum` in toc.html, I have only `lorem ipsum` in it.

How to have `lorem ipsum` below the standard TOC.

Thank you

**Comment by alexvoss** [comment, marked as answer]
You need to copy the `toc.html` from your installation of Materials or you can find it [on GitHub](https://github.com/squidfunk/mkdocs-material/tree/master/src/templates/partials) but make sure to use a version that matches the one you have installed - in case the template changed in ways that are incompatible. Once you have the partial in your `overrides` you can set about customizing it.

**Reply by aborruso** [reply]
Thank you very much

**Reply by aborruso** [reply]
@alexvoss is there a way to not hardcode the `toc.html` source file?

Is there not a way to write something like this?

```
{% extends "toc.html" %}

Lorem ipsum
```

Thank you

**Reply by alexvoss** [reply]
It is possible to [override blocks](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks) but it does not look like the `toc.html`and `toc-item.html` partials define any. If you can say something more about what you are trying to achieve then we might be able to give more specific advice.

**Reply by aborruso** [reply]
> It is possible to [override blocks](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks) but it does not look like the `toc.html`and `toc-item.html` partials define any. If you can say something more about what you are trying to achieve then we might be able to give more specific advice.

I want to add a custom HTML after the TOC: a small sentence with a link that opens a new github issue (capturing the URL of the source page). 

![gg](https://github.com/squidfunk/mkdocs-material/assets/30607/51eba0c2-76ec-4f2b-b7f3-8fef7dda5692)

**Reply by alexvoss** [reply]
Thanks. I think that you should override the partial for now. Perhaps it would be an idea to include an empty block in the partial so people can append things without copying partials. @squidfunk might have a view on that?

**Reply by kamilkrzyskow** [reply]
I think adding an empty block to the partial wouldn't help much. All other blocks are in the `base.html`, which is being extended by `main.html` where all of the block overrides are placed. To allow for blocks in partials, I think it's required to create a rather crazy setup, as in, for each included partial add 1 partial that extends that partial 😈. 
Just like there is base.html and main.html, we could have toc.html and toc_main.html, each include would point to toc_main.html and it would extend toc.html, but this is a rather redundant and as mentioned previously crazy setup 😅 I didn't try it myself, but perhaps it could even work.

As for another way there is of course my guide here: https://github.com/squidfunk/mkdocs-material/discussions/5060
It's a bit outdated, but the idea is the same. I'm postponing updating the guide, as I might create a plugin instead of the hooks approach... but there is always other stuff to do.
The newest **unreleased** hooks, are available in this temporary fork branch https://github.com/kamilkrzyskow/gmc/tree/upgrade-material-9.4/overrides/.hooks

If you need more guidance, I guess, I could spare a few minutes on elaborating @aborruso :v:

**Reply by aborruso** [reply]
> If you need more guidance, I guess, I could spare a few minutes on elaborating @aborruso ✌️

It is too complex for me. For the time being, I use the hard coded copy of the toc.html file.

Thank you very much

**Reply by squidfunk** [reply]
> Perhaps it would be an idea to include an empty block in the partial so people can append things without copying partials

It is not possible to define Jinja blocks in partials. That's just not supported by Jinja. Additionally, we're not considering adding more and more partials for users to hook into. We need to keep the number of partials at a sane level, or build times will go up for everyone.

**Reply by alexvoss** [reply]
I just played with jinja2 and I can see now that adding a block to a partial and overriding it in a `main.html` is [not possible](https://github.com/pallets/jinja/issues/1604), though it seems to be a popular request that once even had an implementation. 

If jinja2 supported this then we would not need additional partials to offer the option to append things to stuff like navigation. This would mean users would not need to copy things around anywhere near as much, which IMHO would help a lot. Even our documentation says that overriding blocks is preferred but the number of actual blocks around where you can do this is limited to the templates, not partials.

Ah well, got to take FOSS as it comes or put in the effort...
